tests/unit/python/foglamp/services/south/test_ingest.py

```diff
@@ -0,0 +1,588 @@
+_current_readings_list_index = 0
+_insert_readings_tasks = None  # type: List[asyncio.Task]
+_readings_list_batch_size_reached = None  # type: List[asyncio.Event]
+_readings_list_not_empty = None  # type: List[asyncio.Event]
+_readings_lists_not_full = None  # type: asyncio.Event
+_insert_readings_wait_tasks = None  # type: List[asyncio.Task]
+_last_insert_time = 0  # type: int
+_readings_list_size = 0  # type: int
+_write_statistics_frequency_seconds = 5
+_readings_buffer_size = 500
+_max_concurrent_readings_inserts = 5
+_readings_insert_batch_size = 100
+_readings_insert_batch_timeout_seconds = 1
+_max_readings_insert_batch_connection_idle_seconds = 60
+_max_readings_insert_batch_reconnect_wait_seconds = 10
+category = 'South'
+default_config = {
+    "write_statistics_frequency_seconds": {
+        "description": "The number of seconds to wait before writing readings-related "
+                       "statistics to storage",
+        "type": "integer",
+        "default": str(_write_statistics_frequency_seconds)
+    },
+    "readings_buffer_size": {
+        "description": "The maximum number of readings to buffer in memory",
+        "type": "integer",
+        "default": str(_readings_buffer_size)
+    },
+    "max_concurrent_readings_inserts": {
+        "description": "The maximum number of concurrent processes that send batches of "
+                       "readings to storage",
+        "type": "integer",
+        "default": str(_max_concurrent_readings_inserts)
+    },
+    "readings_insert_batch_size": {
+        "description": "The maximum number of readings in a batch of inserts",
+        "type": "integer",
+        "default": str(_readings_insert_batch_size)
+    },
+    "readings_insert_batch_timeout_seconds": {
+        "description": "The number of seconds to wait for a readings list to reach the "
+                       "minimum batch size",
+        "type": "integer",
+        "default": str(_readings_insert_batch_timeout_seconds)
+    },
+    "max_readings_insert_batch_connection_idle_seconds": {
+        "description": "Close storage connections used to insert readings when idle for "
+                       "this number of seconds",
+        "type": "integer",
+        "default": str(_max_readings_insert_batch_connection_idle_seconds)
+    },
+    "max_readings_insert_batch_reconnect_wait_seconds": {
+        "description": "The maximum number of seconds to wait before reconnecting to "
+                       "storage when inserting readings",
+        "type": "integer",
+        "default": str(_max_readings_insert_batch_reconnect_wait_seconds)
+    },
+}
+
+@asyncio.coroutine
+def mock_coro():
+    yield from false_coro()
+
+async def false_coro():
+    return True
+
+async def get_cat(old_config):
+    new_config = {}
+    for key, value in old_config.items():
+        new_value = copy.deepcopy(value)
+        new_value['value'] = new_value['default']
+        new_config.update({key: new_value})
+    return new_config
```

Comment:
why? should not we get this as patched config manager?

Reply:
this is better